tests/conftest.py

```diff
@@ -0,0 +1,245 @@
+                        "id": chunk_id,
+                        "model": model,
+                        "created": 1,
+                    }
+                )
+
+        def __aiter__(self):
+            return self
+
+        async def __anext__(self):
+            if self.responses:
+                return ChatCompletionChunk.model_validate(self.responses.pop(0))
+            else:
+                raise StopAsyncIteration
+
+    async def mock_acreate(*args, **kwargs):
+        messages = kwargs["messages"]
+        last_question = messages[-1]["content"]
+        if last_question == "Generate search query for: What is the capital of France?":
+            answer = "capital of France"
+        elif last_question == "Generate search query for: Are interest rates high?":
+            answer = "interest rates"
+        elif isinstance(last_question, list) and last_question[2].get("image_url"):
+            answer = "From the provided sources, the impact of interest rates and GDP growth on "
+            "financial markets can be observed through the line graph. [Financial Market Analysis Report 2023-7.png]"
+        else:
+            answer = "The capital of France is Paris. [Benefit_Options-2.pdf]."
+            if messages[0]["content"].find("Generate 3 very brief follow-up questions") > -1:
+                answer = "The capital of France is Paris. [Benefit_Options-2.pdf]. <<What is the capital of Spain?>>"
+        if "stream" in kwargs and kwargs["stream"] is True:
+            return AsyncChatCompletionIterator(answer)
+        else:
+            return ChatCompletion(
+                object="chat.completion",
+                choices=[
+                    Choice(
+                        message=ChatCompletionMessage(role="assistant", content=answer), finish_reason="stop", index=0
+                    )
+                ],
+                id="test-123",
+                created=0,
+                model="test-model",
+            )
+
+    def patch():
+        monkeypatch.setattr(openai.resources.chat.completions.AsyncCompletions, "create", mock_acreate)
+
+    return patch
+
+
+@pytest_asyncio.fixture(scope="function")
+async def test_client(
+    monkeypatch, app, mock_default_azure_credential, mock_openai_embedding, mock_openai_chatcompletion
+):
+    """Create a test client."""
+    mock_openai_embedding()
+    mock_openai_chatcompletion()
+    with TestClient(app) as test_client:
+        yield test_client
+
+
+@pytest_asyncio.fixture(scope="function")
+async def db_session():
+    """Create a new database session with a rollback at the end of the test."""
+    engine = await create_postgres_engine_from_env()
+    async_sesion = async_sessionmaker(autocommit=False, autoflush=False, bind=engine)
+    session = async_sesion()
+    await session.begin()
+    yield session
+    await session.rollback()
+    await session.close()
+    await engine.dispose()
```

Comment:
this is not used but it can be used for adding more tests in the future to validate the responses directly from the database.